lang/test/org/partiql/lang/eval/EvaluatingCompilerExceptionsTest.kt

```diff
@@ -242,4 +242,37 @@ class EvaluatingCompilerExceptionsTest : EvaluatorTestBase() {
                 Property.COLUMN_NUMBER to 11L,
                 Property.BINDING_NAME to "leading")
         )
+
+    @Test
+    fun intAsNonTextStructField() =
+        checkInputThrowingEvaluationException(
+            "{'apple' : 1, `banana` : 2, 1 : 3}",
+            ErrorCode.EVALUATOR_NON_TEXT_STRUCT_FIELD,
+            mapOf(
+                Property.LINE_NUMBER to 1L,
+                Property.COLUMN_NUMBER to 29L
+            )
+        )
+
+    @Test
+    fun timestampAsNonTextStructField() =
+        checkInputThrowingEvaluationException(
+            "{`2007-02-23T12:14Z` : 3}",
+            ErrorCode.EVALUATOR_NON_TEXT_STRUCT_FIELD,
+            mapOf(
+                Property.LINE_NUMBER to 1L,
+                Property.COLUMN_NUMBER to 2L
+            )
+        )
+
+    @Test
+    fun variableReferenceToIntAsNonTextStructField() =
+        checkInputThrowingEvaluationException(
+            "SELECT {a : 2} FROM {'a' : 1}",
+            ErrorCode.EVALUATOR_NON_TEXT_STRUCT_FIELD,
+            mapOf(
+                Property.LINE_NUMBER to 1L,
+                Property.COLUMN_NUMBER to 8L
```

Comment:
There's some slight inconsistency with the error location. If the non-text struct field name is caught at compile time (as in the previous two tests), the error points to the field name. Whereas, in this case it points to the start of the struct constructor (L8). I think both cases should be consistent with the error pointing to the problematic struct field name.

Reply:
`metas` are not available with the `ExprValue` once it is evaluated.